### DISCUSSION TITLE
Better error handling?

### DISCUSSION THREAD

**Discussion by treddy73** [go-chi/chi#763] [Q&A]
Having some Express experience, I'm wondering if Chi has some kind of error handling that the middleware handles?  If not, can somebody share their design patterns to make the error handling less boilerplate?

**Comment by VojtechVitek** [comment]
What do you need to do? Handle non-200 responses in some middleware?

**Reply by treddy73** [reply]
Since I wrote this question months ago, I've become more comfortable building `http.HandlerFunc` that encapsulates the possible errors and takes advantage of the `render.Renderer` interface.  It took a little time to grok the examples, but they did help me establish some pretty solid patterns. 

For instance:
```
func getEvent(s Service) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {
		e, err := s.Get(chi.URLParam(r, "eventId"))
		if err != nil {
			write(w, r, errNotFound)
			return
		}
		write(w, r, &eventResponse{e})
	}
}
```

My write method is simple:
```
func write(w http.ResponseWriter, r *http.Request, v render.Renderer) {
	if e := render.Render(w, r, v); e != nil {
		_ = render.Render(w, r, errInternalServerError(e))
	}
}
```
